### PR TITLE
SHARE-324 Report Rework

### PR DESCRIPTION
**Pull request by wolfianer**
Added report project to new api.

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing, if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch

### CODE REVIEW

File: catroweb.yaml [hunk 1]
```diff
@@ -1032,6 +1057,19 @@ components:
       items:
         $ref: '#/components/schemas/ProjectResponse'
 
+    ProjectReportResponse:
```
Comment (dmetzner) [inline catroweb.yaml]: typo: should be a Request object

File: catroweb.yaml [hunk 2]
```diff
@@ -1032,6 +1057,19 @@ components:
...
+      type: object
+      properties:
+        user_id:
+          type: string
+          example: 1234abcd-12ab-12ab-12ab-123456abcdef
+        project_id:
```
Comment (dmetzner) [inline catroweb.yaml]: Here the project id is already part of the path: `projects/{ID}/report`

No need for it to be sent a second time in the body

File: catroweb.yaml [hunk 3]
```diff
@@ -1032,6 +1057,19 @@ components:
       items:
         $ref: '#/components/schemas/ProjectResponse'
 
+    ProjectReportResponse:
+      type: object
+      properties:
+        user_id:
```
Comment (dmetzner) [inline catroweb.yaml]: why do we need a user id in this request?

project owner is available through the project

and the reporter (a user) is available through the JWT token. User must be authorized to report

File: catroweb.yaml [hunk 4]
```diff
@@ -450,6 +450,31 @@ paths:
...
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ProjectReportResponse'
+      responses:
+        '204':
+          description: Project successfully reported.
+        '400':
+          $ref: '#/components/responses/BadRequest'
```
Comment (dmetzner) [inline catroweb.yaml:502]: Add  response code for missing/failed authentification

```
401 | Invalid JWT token \| JWT token not found \| JWT token expired
```

**Review by dmetzner — CHANGES_REQUESTED** [review]

File: catroweb.yaml [hunk 5]
```diff
@@ -1032,6 +1057,19 @@ components:
...
+      properties:
+        user_id:
+          type: string
+          example: 1234abcd-12ab-12ab-12ab-123456abcdef
+        project_id:
+          type: string
+          example: 1234abcd-12ab-12ab-12ab-123456abcdef
+        category:
+          type: string
+          example: Graphic violence
```
Comment (HCrane) [inline catroweb.yaml]: This should be an enum with the categories provided by the report dialogue.

**Review by HCrane — CHANGES_REQUESTED** [review]

**Review by HCrane — APPROVED** [review]
👍

**Review by dmetzner — APPROVED** [review]

**Pull request merged**